dvs10_Options.m

```diff
@@ -0,0 +1,883 @@
+f5.Target = [999 999];                  % can't be chosen             
+f5.Threshold = hold_opt;
+
+%% TASK
+set_bgcolor(bg_trial);      % set trial background
+dashboard(1, sprintf('Number trials initiated: %0.0f',TrialRecord.User.num_initiated_trials), white);
+dashboard(2, sprintf('Number trials attempted: %0.0f',TrialRecord.User.num_choicemade_trials), white);
+dashboard(3, sprintf('Performance: %0.0f%% correct',TrialRecord.User.perc_correct_trials), white)
+
+% save trial variables 
+if x_position1_i==1                                     % option 1 is on the left
+    dvs_left_value  = [dvs_left_value option1_value];
+    dvs_left_mag    = [dvs_left_mag mag1_i];
+    dvs_left_prob   = [dvs_left_prob prob1_i];
+    dvs_right_value = [dvs_right_value option2_value];
+    dvs_right_mag   = [dvs_right_mag mag2_i];
+    dvs_right_prob  = [dvs_right_prob prob2_i];
+elseif x_position1_i==2                                 % option 1 is on the right
+    dvs_left_value  = [dvs_left_value option2_value];
+    dvs_left_mag    = [dvs_left_mag mag2_i];
+    dvs_left_prob   = [dvs_left_prob prob2_i];
+    dvs_right_value = [dvs_right_value option1_value];
+    dvs_right_mag   = [dvs_right_mag mag1_i];
+    dvs_right_prob  = [dvs_right_prob prob1_i];
+end
+bhv_variable('dvs_left_value', dvs_left_value);
+bhv_variable('dvs_left_mag', dvs_left_mag);
+bhv_variable('dvs_left_prob', dvs_left_prob);
+bhv_variable('dvs_right_value', dvs_right_value);
+bhv_variable('dvs_right_mag', dvs_right_mag);
+bhv_variable('dvs_right_prob', dvs_right_prob);
+bhv_variable('stimulus_set',stimulus_set);
+
+%% ITI
+% start trial with blank screen = baseline/pre-fixation epoch
+set_bgcolor(bg_trial)
+eventmarker(ec_iti_start)
+idle(i ti)
+eventmarker(ec_iti_end)
+
+%% 1. FIXATION
+% need to fixate centre (fixation breaks allowed). if no fixation acquired
+% trial is stopped, error screen and timeout
+run_scene(fixation_scene,ec_fix_screen_shown); 
+
+    % no fixation
+    if ~fixth.Success
+            trialerror(ec_out_nofix)
+            idle(nofix_timeout,bg_nofix)
+            eventmarker(ec_timeout_screenshown)
+    end
+    
+
+%% 2. OPTION 1 ONSET
+% option 1 is shown. no fixation required to continue. record if/which cues
+% are fixated.
+if fixth.Success
+TrialRecord.User.num_initiated_trials = TrialRecord.User.num_initiated_trials+1;
+
+option1_onset_time = run_scene(option1_scene,ec_option1_on);
+  
+    if mag_fth1.Success
+        eventmarker(ec_mag1_fix)
+        if x_position1_i==1     % option 1 on the left
+            if mag1_y_position_i==1     % mag is up
+                eventmarker(ec_upleft_fix)
+            else
+                eventmarker(ec_downleft_fix)
+            end
+        else
+            if mag1_y_position_i==1    
+                eventmarker(ec_upright_fix)
+            else
+                eventmarker(ec_downright_fix)
+            end
+        end
+
+        time_mag1_acquired = mag_fth1.AcquiredTime;                              

```

Comment:
The one thing I don't know about FTH is whether repeated fixations can also occur after `Success=1`, which would overwrite the earlier fixations. If that is the case, this would mean you are capturing the last fixation only, so if during cue presentation they go mag cue -> prob cue -> mag cue then you will get the time of the second fixation for the mag cue (And the wrong order of fixation). 

I think a WaitThenHold doesn't suffer from this, it will only log the first fixation, and first fixation only. So I would do a WTH with a fixation time of like 50 ms to avoid any false positives. 

But you just need to be aware you will be blind to any multiple fixations of a target (probably okay given everything else you have on your plate..)

Reply:
And now I see you have each option on for 800 ms. That is ages. They probably will be flicking between them (i guess you already know as you watch them play). I guess the later fixations are very unlikely to actually be of interest, but definitely make sure you store information about the first fixations, which is when the value signals should be occurring

Reply:
So definitely very important to test this (or maybe you have already done so and know the answer). Just put the fixation time to something like 10 seconds, run task on simulation mode, fixate the target immediately, then again after 5 seconds and check what fixation time is recorded